src/spine.sig

```diff
@@ -1,3 +1,10 @@
+(* A spine is an abstraction over the sequences of symbols and variables
+ * that occur in binders. So that [x.y.] portion of [natrec(a; b; x.y.c)]
+ *
+ * The extra level of generality given by making t polymorphic means that
+ * the same sort of spine can be used to store both symbols and variables for
+ * example.
```

Comment:
- [ ] same as above, we have spines of terms, spines of abstractions, spines of sorts, spines of valences, spines of variables, spines of symbols, etc.